tests/tracer/test_processors.py

```diff
@@ -329,3 +336,108 @@ def test_span_normalizator():
+
+
+def test_single_span_sampling_processor_match_second_rule():
+    """Test that single span sampling rule is applied if the first rule does not match, but a later one does"""
+
+    rule_1 = SpanSamplingRule(service="test_service", name="test_name")
+    rule_2 = SpanSamplingRule(service="test_service2", name="test_name2")
+    rules = [rule_1, rule_2]
+    processor = SpanSamplingProcessor(rules)
+    tracer = Tracer()
+    tracer.configure(writer=DummyWriter())
+    tracer._span_processors.append(processor)
+
+    span = traced_function(tracer, name="test_name2", service="test_service2")
+
+    assert_sampling_decision_tags(span)
+
+
+def test_single_span_sampling_processor_rule_order_drop():
+    """Test that single span sampling rules are applied in an order and
+    will only be applied if earlier rules have not been
+    """
+
+    rule_1 = SpanSamplingRule(service="test_service", name="test_name", sample_rate=0)
+    rule_2 = SpanSamplingRule(service="test_service", name="test_name", sample_rate=1.0)
+    rules = [rule_1, rule_2]
+    processor = SpanSamplingProcessor(rules)
+    tracer = Tracer()
+    tracer.configure(writer=DummyWriter())
+    tracer._span_processors.append(processor)
+
+    span = traced_function(tracer)
+
+    assert_sampling_decision_tags(span, sample_rate=None, mechanism=None, limit=None)
+
+
+def test_single_span_sampling_processor_rule_order_keep():
+    """Test that single span sampling rules are applied in an order
+    and will not be applied if an earlier rule has been
+    """
+
+    rule_1 = SpanSamplingRule(service="test_service", name="test_name", sample_rate=1.0)
+    rule_2 = SpanSamplingRule(service="test_service", name="test_name", sample_rate=0)
+    rules = [rule_1, rule_2]
+    processor = SpanSamplingProcessor(rules)
+    tracer = Tracer()
+    tracer.configure(writer=DummyWriter())
+    tracer._span_processors.append(processor)
+
+    span = traced_function(tracer)
+
+    assert_sampling_decision_tags(span)
+
+
+def test_single_span_sampling_processor_do_not_tag_if_tracer_samples():
+    """Test that single span sampling rules aren't applied if a span is already going to be sampled by trace sampler"""
+
+    rule_1 = SpanSamplingRule(service="test_service", name="test_name")
+    rules = [rule_1]
+    processor = SpanSamplingProcessor(rules)
+    tracer = Tracer()
+    tracer.configure(writer=DummyWriter())
+    tracer._span_processors.append(processor)
+
+    span = traced_function(tracer, trace_sampling=True)
+
+    assert_sampling_decision_tags(span, sample_rate=None, mechanism=None, limit=None, trace_sampling=True)
+
+
+def traced_function(tracer, name="test_name", service="test_service", trace_sampling=False):
+    with tracer.trace(name) as span:
+        # If the trace sampler samples the trace, then we shouldn't add the span sampling tags
+        if trace_sampling:
+            span.context.sampling_priority = 1
+        else:
+            span.context.sampling_priority = 0
```

Comment:
Did you mean to set the priority on the trace to drop/reject?

```suggestion
            span.context.sampling_priority = -1
```

Reply:
Actually, we should cover both user and auto drop.

Reply:
Yes, doesn't 0 drop?